linux_os/guide/system/accounts/accounts-session/accounts_user_interactive_home_directory_defined/oval/shared.xml

```diff
@@ -0,0 +1,36 @@
+<def-group>
+  <definition class="compliance" id="{{{ rule_id }}}" version="1">
+    {{{ oval_metadata("All Interactive Users Must Have A Home Directory Defined") }}}
+    <criteria>
+      <criterion test_ref="test_accounts_user_interactive_home_directory_defined"
+                 comment="All Interactive Users Must Have A Home Directory Defined"/>
+    </criteria>
+  </definition>
+
+  <!-- For detailed comments about logic used in this OVAL, check the
+       "file_ownership_home_directories" rule.
+       #### creation of object #### -->
+  <unix:password_object id="object_accounts_user_interactive_home_directory_defined_objects"
+                        version="1">
+    <unix:username datatype="string" operation="not equal">nobody</unix:username>
+    <filter action="include">state_accounts_user_interactive_home_directory_defined_uids</filter>
+  </unix:password_object>
+
+  <unix:password_state id="state_accounts_user_interactive_home_directory_defined_uids"
+                       version="1">
+    <unix:user_id datatype="int" operation="greater than or equal">{{{ uid_min }}}</unix:user_id>
```

Comment:
Is it clear what is an interactive user? Does STIG define it?

https://vaulted.io/library/disa-stigs-srgs/red_hat_enterprise_linux_8_security_technical_implementation_guide/V-230320
The STIG item related to this rule doesn't explicitly say to check `root` user, but defines a fallback to check users from UID 1000.

But int https://vaulted.io/library/disa-stigs-srgs/red_hat_enterprise_linux_8_security_technical_implementation_guide/V-230323 it seems to define interactive users the ones that can login and leave logon traces.

Reply:
Hi @yuumasato , this is not explicit neither on CIS. However, I am much more prone to interpret `interactive users` as `regular users`. The fallback you mentioned also suggests that STIG interprets this way too.

Considering the rules are related to home directories "definition" and "existence" I don't think it makes sense to test root home directory too, since it should always be defined and present in any distro which follows the Filesystem Hierarchy Standard (FHS).

I would add that if we need the same tests for root (not a regular user), separate rules should be created because even the logic to assess it is a little bit different.